src/dxe_services.rs

```diff
@@ -0,0 +1,317 @@
+pub type AddIoSpace = extern "efiapi" fn(GcdIoType, PhysicalAddress, u64) -> Status;
+
+/// This service allocates nonexistent I/O, reserved I/O, or I/O resources from the global coherency domain of the processor.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.2.4.10
+pub type AllocateIoSpace =
+  extern "efiapi" fn(GcdAllocateType, GcdIoType, u32, u64, *mut PhysicalAddress, Handle, Handle) -> Status;
+
+/// This service frees nonexistent I/O, reserved I/O, or I/O resources from the global coherency domain of the processor.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.2.4.11
+pub type FreeIoSpace = extern "efiapi" fn(PhysicalAddress, u64) -> Status;
+
+/// This service removes reserved I/O, or I/O resources from the global coherency domain of the processor.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.2.4.12
+pub type RemoveIoSpace = extern "efiapi" fn(PhysicalAddress, u64) -> Status;
+
+/// This service retrieves the descriptor for an I/O region containing a specified address.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.2.4.13
+pub type GetIoSpaceDescriptor = extern "efiapi" fn(PhysicalAddress, *mut IoSpaceDescriptor) -> Status;
+
+/// Returns a map of the I/O resources in the global coherency domain of the processor.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.2.4.14
+pub type GetIoSpaceMap = extern "efiapi" fn(*mut usize, *mut *mut IoSpaceDescriptor) -> Status;
+
+/// Loads and executes DXE drivers from firmware volumes.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.3.1
+pub type Dispatch = extern "efiapi" fn() -> Status;
+
+/// Clears the Schedule on Request (SOR) flag for a component that is stored in a firmware volume.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.3.2
+pub type Schedule = extern "efiapi" fn(Handle, *const Guid) -> Status;
+
+/// Promotes a file stored in a firmware volume from the untrusted to the trusted state.
+/// Only the Security Architectural Protocol can place a file in the untrusted state.
+/// A platform specific component may choose to use this service to promote a previously untrusted file to the trusted state.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.3.3
+pub type Trust = extern "efiapi" fn(Handle, *const Guid) -> Status;
+
+/// Creates a firmware volume handle for a firmware volume that is present in system memory.
+///
+/// # Documentation
+/// UEFI Platform Initialization Specification, Release 1.8, Section II-7.3. II-59 (This one does not have a section)
+pub type ProcessFirmwareVolume = extern "efiapi" fn(*const c_void, u32, *mut Handle) -> Status;
```

Comment:
There are places like `ProcessFirmwareVolume()` that need to be scrubbed for using `u32` instead of `usize` for `UINTN` values.

```suggestion
pub type ProcessFirmwareVolume = extern "efiapi" fn(*const c_void, usize, *mut Handle) -> Status;
```

Reply:
Done in 81888349dcd592444fc8f98af261e0cdf9741fb8